test/unit/carriers/fedex_test.rb

```diff
@@ -205,6 +205,31 @@ def test_parsing_response_with_no_rate_reply
     assert_equal message, exception.message
   end
 
+  def test_parsing_response_with_no_system_code_for_one_rate_does_not_crash
+    mock_response = xml_fixture('fedex/freight_rate_response').gsub('<v13:ServiceType>FEDEX_FREIGHT_ECONOMY</v13:ServiceType>','')
+
+    @carrier.expects(:commit).returns(mock_response)
+    @carrier.logger = Logger.new(StringIO.new)
+    @carrier.logger.expects(:warn)
```

Comment:
Thanks for adding that, if you would like to be more explicit you can add:
`@carrier.logger.expects(:warn).with(mock_response).once`